pennylane/transforms/decompositions/clifford_t/clifford_t_transform.py

```diff
@@ -0,0 +1,497 @@
+
+    copied_ops = operations.copy()
+    merged_ops, number_ops = [], 0
+
+    while len(copied_ops) > 0:
+        curr_gate = copied_ops.pop(0)
+
+        # If gate is not to be merged, let it go
+        if merge_ops is not None and curr_gate.name not in merge_ops:
+            merged_ops.append(curr_gate)
+            continue
+
+        # If gate is in the merge_ops, update counter
+        if curr_gate.name in merge_ops:
+            number_ops += 1
+
+        # Find the next gate that acts on the same wires
+        next_gate_idx = find_next_gate(curr_gate.wires, copied_ops)
+        if next_gate_idx is None:
+            merged_ops.append(curr_gate)
+            continue
+
+        # Initialize the current angle and iterate until end of merge
+        curr_params = curr_gate.parameters
+        curr_intrfc = qml.math.get_deep_interface(curr_gate.parameters)
+        cumulative_angles = qml.math.array(curr_params, dtype=float, like=curr_intrfc)
+        next_gate = copied_ops[next_gate_idx]
+        while curr_gate.name == next_gate.name and curr_gate.wires == next_gate.wires:
+            cumulative_angles += qml.math.array(next_gate.parameters, like=curr_intrfc)
+            # Check if the subsequent gate exists in the vicinity
+            copied_ops.pop(next_gate_idx)
+            next_gate_idx = find_next_gate(curr_gate.wires, copied_ops)
+            if next_gate_idx is None:
+                break
+            next_gate = copied_ops[next_gate_idx]
+
+        # Replace the current gate, add it to merged list and pop it from orginal one
+        merged_ops.append(curr_gate.__class__(*cumulative_angles, wires=curr_gate.wires))
+
+    return merged_ops, number_ops
+
+
+# pylint: disable= too-many-nested-blocks, too-many-branches, too-many-statements, unnecessary-lambda-assignment
+@transform
+def clifford_t_decomposition(
+    tape: QuantumTape,
+    epsilon=1e-4,
+    max_expansion=6,
+    method="sk",
+    **method_kwargs,
+) -> (Sequence[QuantumTape], Callable):
+    r"""Decomposes a circuit into the Clifford+T basis.
+
+    This method first decomposes the gate operations to a basis comprised of Clifford, :class:`~.T`, :class:`~.RZ` and
+    :class:`~.GlobalPhase` operations (and their adjoints). The Clifford gates include the following PennyLane operations:
+
+    - Single qubit gates - :class:`~.Identity`, :class:`~.PauliX`, :class:`~.PauliY`, :class:`~.PauliZ`,
+      :class:`~.SX`, :class:`~.S`, and :class:`~.Hadamard`.
+    - Two qubit gates - :class:`~.CNOT`, :class:`~.CY`, :class:`~.CZ`, :class:`~.SWAP`, and :class:`~.ISWAP`.
+
+    Then the leftover single qubit :class:`~.RZ` operations are approximated in the Clifford+T basis with
+    :math:`\epsilon > 0` error. By default, we use the Solovay-Kitaev algorithm described in
+    `Dawson and Nielsen (2005) <https://arxiv.org/abs/quant-ph/0505030>`_ for this.
+
+    Args:
+        tape (QNode or QuantumTape or Callable): The quantum circuit to be decomposed.
```

Comment:
I can't get this to work as a qfunc:
```python
import pennylane as qml

def f():
    qml.CRY(0.5, [0, 1])

qml.clifford_t_decomposition(f)()
```

Reply:
works for me, you just need to call it in a queuing context

Reply:
fwiw, I think this is a matter of opinion. qfuncs probably _should_ only be used in recording contexts because all they do is queue operators so they're totally useless outside of queuing contexts. I think some users will do as you did anyway, but imo I would leave the error - it's better to raise an error than to let the user believe something actually happened when it didn't

Reply:
@timmysilv could you share the code you recommend to get this to work?

Reply:
```python
import pennylane as qml

def f():
    qml.CRY(0.5, [0, 1])

tape = qml.tape.make_qscript(qml.clifford_t_decomposition(f))()
```

or:
```python
with qml.queuing.AnnotatedQueue() as q:
    qml.clifford_t_decomposition(f)()
```

Reply:
Thanks! Yep this is great.